code/modules/asset_cache/assets/crafting.dm

```diff
@@ -2,33 +2,48 @@
 /datum/asset/spritesheet/crafting
 	name = "crafting"
 
-/datum/asset/spritesheet/crafting/cooking
-	name = "cooking"
-
 /datum/asset/spritesheet/crafting/create_spritesheets()
 	var/id = 1
 	for(var/atom in GLOB.crafting_recipes_atoms)
 		add_atom_icon(atom, id++)
 	add_tool_icons()
 
+/datum/asset/spritesheet/crafting/cooking
+	name = "cooking"
+
 /datum/asset/spritesheet/crafting/cooking/create_spritesheets()
 	var/id = 1
 	for(var/atom in GLOB.cooking_recipes_atoms)
 		add_atom_icon(atom, id++)
 
-///Adds the atom icon to the spritesheet with given ID
-/datum/asset/spritesheet/crafting/proc/add_atom_icon(atom, id)
-	var/obj/obj = initial(atom)
-	if(ispath(atom, /datum/reagent))
-		var/datum/reagent/reagent = atom
-		obj = initial(reagent.default_container)
-	var/icon_file = initial(obj.icon_preview) || initial(obj.icon)
-	var/icon_state = initial(obj.icon_state_preview) || initial(obj.icon_state)
-	#ifdef UNIT_TESTS
-	if(!(icon_state in icon_states(icon_file)))
-		stack_trace("Atom [atom] with icon '[icon_file]' missing state '[icon_state]'")
+/**
+ * Adds the ingredient icon to the spritesheet with given ID
+ *
+ * ingredient_typepath can be an obj typepath OR a reagent typepath
+ *
+ * If it a reagent, it will use the default container's icon state,
+ * OR if it has a glass style associated, it will use that
+ */
+/datum/asset/spritesheet/crafting/proc/add_atom_icon(ingredient_typepath, id)
+	var/icon_file
+	var/icon_state
+	var/obj/preview_item = ingredient_typepath
+	if(ispath(ingredient_typepath, /datum/reagent))
+		var/datum/reagent/reagent = ingredient_typepath
+		preview_item = initial(reagent.default_container)
+		var/datum/glass_style/style = GLOB.glass_style_singletons[preview_item]?[reagent]
+		if(istype(style))
+			icon_file = style.icon
+			icon_state = style.icon_state
+
+	icon_file ||= initial(preview_item.icon_preview) || initial(preview_item.icon)
+	icon_state ||= initial(preview_item.icon_state_preview) || initial(preview_item.icon_state)
+
+#ifdef UNIT_TESTS
+	if(!icon_exists(icon_file, icon_state, scream = TRUE))
 		return
-	#endif
+#endif
```

Comment:
Couldn't you change this to be a focus-only unit test with `PERFORM_ALL_TESTS`? Would be a lot more specific/self-documenting that this is a "testing" check, and not some assertion we pass in here for when we're running UNIT_TESTS.

Reply:
feels like a good idea to change it over imho